File: variant-b-old-hermes/android/app/src/main/java/com/dvrn/app/DebugModule.java
package com.dvrn.app;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;

public class DebugModule extends ReactContextBaseJavaModule {

  DebugModule(ReactApplicationContext reactContext) {
    super(reactContext);
  }

  @Override
  public String getName() {
    return "DebugModule";
  }

  @ReactMethod
  public void readInternalFile(String path, Promise promise) {
    try {
      byte[] bytes = Files.readAllBytes(new File(path).toPath());
      promise.resolve(new String(bytes));
    } catch (IOException e) {
      promise.reject("READ_ERROR", e);
    }
  }
}

File: variant-a-no-hermes/android/app/src/main/java/com/dvrn/app/DebugModule.kt
package com.dvrn.app

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import java.io.File

class DebugModule(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

  override fun getName() = "DebugModule"

  @ReactMethod
  fun readInternalFile(path: String, promise: Promise) {
    try {
      val content = File(path).readText()
      promise.resolve(content)
    } catch (e: Exception) {
      promise.reject("READ_ERROR", e)
    }
  }
}
